spring-xd-spark-streaming/src/main/java/org/springframework/xd/spark/streaming/Processor.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.xd.spark.streaming;
+
+import java.io.Serializable;
+
+import org.apache.spark.streaming.api.java.JavaDStreamLike;
+
+/**
+ * Interface for modules using the Spark Streaming API to process messages from the Message Bus.
+ *
+ * @author Mark Fisher
+ */
+public interface Processor extends Serializable {
+
+	public static final String SPARK_MASTER_URL_PROP = "spark.master";
```

Comment:
That was the original idea behind XD-2465, but now I think the plan is to have the Reactor and RxJava module support follow a similar plugin-based approach (with a module execution framework indicator) to what we're doing here with Spark Streaming. That will be on the roadmap after the 1.1 release.